### PR TITLE
[imager] Move droid.info generation to oneshot. JB#50581

### PR DESCRIPTION
**Pull request by abranson**
It won't work in the imager, and should be postponed until first boot.

### CODE REVIEW

File: rpm/gmp-generate-info.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+set -e
+if [ "$MIC_RUN" != "" ]; then
+	echo "gmp-generate-info.sh.sh - returning FAIL to postpone oneshot to first boot"
```
Comment (monich) [inline rpm/gmp-generate-info.sh:4]: .sh.sh

Comment (abranson) [reply rpm/gmp-generate-info.sh]: Pocket sand removed :)

File: rpm/gmp-droid.spec [hunk 2]
```diff
@@ -24,9 +26,15 @@ Gecko Media Plugin for droidmedia codec support in Gecko based browsers
 %install
 %meson_install
 
+# create config oneshot
+mkdir -p $RPM_BUILD_ROOT%{_oneshotdir}
+cp %{SOURCE1} $RPM_BUILD_ROOT%{_oneshotdir}/
```
Comment (okodron) [inline rpm/gmp-droid.spec:31]: You can use `install  -D -m 0755 %{SOURCE1} %{buildroot}/%{_oneshotdir}` instead.

File: rpm/gmp-droid.spec [hunk 3]
```diff
@@ -7,7 +7,9 @@ Release:        1
 License:        MPL2.0
 URL:            https://github.com/sailfishos/%{name}
 Source0:        %{name}-%{version}.tar.gz
+Source1:        gmp-generate-info.sh
 BuildRequires:  meson
+BuildRequires:  oneshot
 BuildRequires:  pkgconfig(libandroid-properties)
 BuildRequires:  droidmedia-devel
 Requires:       droidmedia
```
Comment (okodron) [inline rpm/gmp-droid.spec:15]: Also necessary:
```
Requires:  oneshot
%{_oneshot_requires_post}
```

File: rpm/gmp-droid.spec [hunk 4]
```diff
@@ -24,9 +26,15 @@ Gecko Media Plugin for droidmedia codec support in Gecko based browsers
...
+mkdir -p $RPM_BUILD_ROOT%{_oneshotdir}
+cp %{SOURCE1} $RPM_BUILD_ROOT%{_oneshotdir}/
+echo "%{_libdir}/xulrunner-qt5-%{gecko_ver}/%{name}/0.1/generate-info 1>%{_libdir}/xulrunner-qt5-%{gecko_ver}/%{name}/0.1/droid.info 2>/dev/null" >> $RPM_BUILD_ROOT%{_oneshotdir}/gmp-generate-info.sh
+chmod +x $RPM_BUILD_ROOT%{_oneshotdir}/gmp-generate-info.sh
+
 %post
-# Query device codec support and write out the droid.info file
-%{_libdir}/xulrunner-qt5-%{gecko_ver}/%{name}/0.1/generate-info 1>%{_libdir}/xulrunner-qt5-%{gecko_ver}/%{name}/0.1/droid.info 2>/dev/null
+# Query device codec support and write out the droid.info file. On imager this should postpone until first boot.
+add-oneshot --now gmp-generate-info.sh
```
Comment (okodron) [inline rpm/gmp-droid.spec:37]: `%{_bindir}/add-oneshot`

File: rpm/gmp-droid.spec [hunk 5]
```diff
@@ -24,9 +28,14 @@ Gecko Media Plugin for droidmedia codec support in Gecko based browsers
 %install
 %meson_install
 
+# create config oneshot
+mkdir -p $RPM_BUILD_ROOT%{_oneshotdir}
```
Comment (okodron) [inline rpm/gmp-droid.spec:32]: `install -D` already creates all the necessary directories in the path.

File: rpm/gmp-droid.spec [hunk 6]
```diff
@@ -24,9 +28,14 @@ Gecko Media Plugin for droidmedia codec support in Gecko based browsers
...
+install -D -m 0755 %{SOURCE1} %{buildroot}/%{_oneshotdir}
+echo "%{_libdir}/xulrunner-qt5-%{gecko_ver}/%{name}/0.1/generate-info 1>%{_libdir}/xulrunner-qt5-%{gecko_ver}/%{name}/0.1/droid.info 2>/dev/null" >> $RPM_BUILD_ROOT%{_oneshotdir}/gmp-generate-info.sh
+
 %post
-# Query device codec support and write out the droid.info file
-%{_libdir}/xulrunner-qt5-%{gecko_ver}/%{name}/0.1/generate-info 1>%{_libdir}/xulrunner-qt5-%{gecko_ver}/%{name}/0.1/droid.info 2>/dev/null
+# Query device codec support and write out the droid.info file. On imager this should postpone until first boot.
+%{_bindir}/add-oneshot --now gmp-generate-info.sh
 
 %files
```
Comment (okodron) [inline rpm/gmp-droid.spec:40]: `%defattr(-,root,root,-)` is missing.

File: rpm/gmp-droid.spec (hunk 5 above)
Comment (abranson) [reply rpm/gmp-droid.spec]: I had to add the full filepath for that to work, but it works. Thanks.

**Pull request merged**